plugins/shared/structs/units.go

```diff
@@ -0,0 +1,213 @@
+package structs
+
+import "sort"
+
+var (
+	// numUnits is the number of known units
+	numUnits = len(binarySIBytes) + len(decimalSIBytes) + len(binarySIByteRates) + len(decimalSIByteRates) + len(watts) + len(hertz)
+
+	// UnitIndex is a map of unit name to unit
+	UnitIndex = make(map[string]*Unit, numUnits)
+
+	// lengthSortedUnits is a list of unit names sorted by length with longest
+	// first
+	lengthSortedUnits = make([]string, 0, numUnits)
+
```

Comment:
I didn;t check the conversion values below too closely, since there's no unit test and we are setting up the units here its worth a second  look

Reply:
Double checked them